pkg/daemon/server/metrics.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2022 The Numaproj Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package server
+
+import (
+	"context"
+	"math"
+	"time"
+
+	"github.com/numaproj/numaflow/pkg/metrics"
+
+	"github.com/numaproj/numaflow/pkg/apis/proto/daemon"
+	"github.com/numaproj/numaflow/pkg/shared/logging"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"go.uber.org/zap"
+)
+
+var (
+	pipelineProcessingLag = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name:      "pipeline_lag_milliseconds",
+		Help:      "pipeline processing lag metrics in milliseconds.",
+		Subsystem: "daemon",
+	}, []string{metrics.LabelPipeline})
+)
+
+var (
+	currentTimeToWatermark = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name:      "current_time_compared_to_watermark_milliseconds",
```

Comment:
We need a better metrics name...

Reply:
@yhl25 - this is the metric you want, right?

Reply:
yes